bin/cogeqc-summarize-ogs.R

```diff
@@ -0,0 +1,194 @@
+            rownames(stats) <- species
+            stats <- cbind(data.frame(Species = rownames(stats)), stats)
+            stats$Species <- as.factor(stats$Species)
+            rownames(stats) <- NULL
+            result_list <- list(stats = stats, og_overlap = og_overlap)
+            return(result_list)
+        }
+
+# Additionally, edit the assess_orthogroups function to run in parallel,
+# using mclappy from the parallel package
+assess_orthogroups <- 
+function (orthogroups = NULL, annotation = NULL, correct_overclustering = TRUE, mc.cores = 1) 
+{
+    og_list <- split(orthogroups, orthogroups$Species)
+    og_list <- mclapply(seq_along(og_list), mc.cores = mc.cores, function(x) {
+        species <- names(og_list)[x]
+        idx <- which(names(annotation) == species)
+        merged <- merge(og_list[[x]], annotation[[idx]])
+        names(merged)[4] <- "Annotation"
+        H <- calculate_H(merged, correct_overclustering = correct_overclustering)
+        names(H) <- c("Orthogroups", paste0(species, "_score"))
+        return(H)
+    })
+    merge_func <- function(x, y) {
+        merge(x, y, by = "Orthogroups", all = TRUE)
+    }
+    final_df <- Reduce(merge_func, og_list)
+    means <- apply(final_df[, -1], 1, mean, na.rm = TRUE)
+    medians <- apply(final_df[, -1], 1, median, na.rm = TRUE)
+    final_df$Mean_score <- means
+    final_df$Median_score <- medians
+    return(final_df)
+}
+
+args = commandArgs(trailingOnly=TRUE)
+
+# Get the base directory to where the orthofinder results are.
+# Again, specified from the commandline
+ogDir <- args[1]
+
+# Specify the path to protein annotations to be used by cogeqc.
+# Again, specified from the commandline
+annotDir <- './' #args[2]
+# annotDir <- './results/protein-annotations/'
+
+# Pull out the inflation parameter from the filepath
+inflation <- gsub(".*_", "", ogDir)
+
+# Now, create a variable using these to specify paths to the orthofinder 
+# orthogroups file
+ogFile <- paste0(ogDir, '/Orthogroups/Orthogroups.tsv')
+
+# Specify the path to orthofinders orthogroup summary stats to be used by cogeqc.
+ogStatDir <- paste0(ogDir, '/Comparative_Genomics_Statistics/')
+
+# Go ahead and read in the orthogroups file
+orthogroups <- read_orthogroups(ogFile)
+
+# Strip trailing text from species name - may not need in full implementation. 
+# Names are determined in orthofinder using the file name, so just include 
+# the species here. 
+orthogroups$Species <- gsub('[.].*', '', orthogroups$Species)
+
+# Pull out the list of species - we are only running this script on a 
+# on a subset of species, so we want to be sure that we're not reading in 
+# annotations for species other than those we're testing inflation 
+# parameters with. 
+species <- unique(orthogroups$Species)
+
+# and remove the Species name from the gene name - this will create issues when
+# pairing with the annotations. 
+orthogroups$Gene <- gsub('^(?:[^_]*_)*\\s*(.*)', '\\1', orthogroups$Gene)
```

Comment:
So I'm making this under the expectation that gene/sequence names will follow a standardized format of Genus_species:GeneID (there can be additional stuff prior to the colon so long as there is no colon) - everything following (after a space) is ignored by orthofinder and won't be included here. This just takes everything after the colon. 

I agree that to make it more generalizable we could accommodate other formats, but so that all of these many methods play well, I figured I'd just take a firm stance on standardizing sequence names - doesn't need to be set in stone though.

Reply:
this probably needs to be in the documentation somewhere for how input files should be formatted if it's expected to be this way

Reply:
Yeah, I can definitely do that. I intend to do a massive overhaul of the readme to clarify everything, including how I expect the input sequences to be formatted, and how the samplesheet should be formatted.

Reply:
Okay, added a full description in the README that should clear this up!